docs/api/api-spec.md

```diff
@@ -132,6 +132,34 @@ On the financial desktop, applications often want to broadcast context to any nu
 ## Resolvers
 Intents functionality is dependent on resolver functionality to map the intent to a specific App.  This will often require end-user input.  Resolution can either be performed by the Desktop Agent (raising UI to pick the desired App for the intent) or by the app launching the intent - in which case the calling App will handle the resolution itself (using the findIntents API below) and then invoke an explicit Intent object.
 
+## Context channels
+
+The context channel api allows a set of apps to share a stateful piece of data between them, and be alerted when it changes.
+
+There are three types of channels, which are functionally identical, but have different visibility and discoverability semantics.
+
+1. The 'public' or 'desktop' ones, which have a well understood identity. One is called 'default'.
+2. The 'private' or 'app' ones, which have a transient identity and need to be revealed
+
+The 'public' channels include a 'default' channel which serves as the backwards compatible layer with the 'send/broadcast context' above. There are some reserved channel names for future use. Currently this is just 'global'.
+
+To find a desktop channel, one calls
```

Comment:
I am trying to understand how apps are going to use this API. Looking forward to some pseudo code demonstrating the three use cases we talked in the meeting.

I would like to know how an app associates itself with a channel or does a DesktopAgent puts an app in a channel (i.e. in the later case app knows nothing about channels). I was imagining some sort of `joinChannel()` api somewhere.

Also how this separate `channel.broadcast()` function is going to work. Is this something an app will call? I think the idea was app doesn't even need to know about channels.  Will an app just use regular top level `fdc3.broadcast()` and the DesktopAgent intercept that and will then use the `channel.broadcast()` to route it accordingly? How are those two separate `broadcast()` play nicely with each other.

Same thing about `addBroadcastListener()` vs `addContextListener()`. What goes in the app vs the desktopAgent?

Hopefully some code around use cases will clear these things.

Reply:
In terms of an app associating itself with a channel, something in the app will call `getDesktopChannels`, will find the channel it needs by checking the members of the `DesktopChannelVisualIdentity` interface, and will squirrel that away, plus the listener it got from subscribing to it.

In terms of the classic `fdc3.broadcast()` vs `channel.broadcast()`: the code is suggesting that one of the desktopChannels will be called 'default', and that if you call `fdc3.broadcast()` you will send to this channel and vice-versa. In other words, that channel is a facade to the existing single global context.